### ISSUE TITLE
Knip doesn't work with a fresh NX(19.4.2) Angular(18) Monorepo 

### ISSUE THREAD

**Issue by AvioLT** [webpro-nl/knip#727]
### Reproduction url

https://nx.dev/getting-started/tutorials/angular-monorepo-tutorial

### Description of the issue

So, I was trying to build up a config slowly to match my NX monorepo structure. I wanted to do this by making a plain new NX project and adding knip to it. However, it fails.

So starting from [https://nx.dev/getting-started/tutorials/angular-monorepo-tutorial](https://nx.dev/getting-started/tutorials/angular-monorepo-tutorial)
I get a plain repo with some basic config (Angular, Jest, Cypress, Webpack, ...)
![image](https://github.com/user-attachments/assets/679a4359-1bd5-452f-b7c1-0b4a501c960c)

This gives me a project with the following versions
```
    "nx": "19.4.2",
    "@angular/core": "~18.0.0",
    "cypress": "^13.13.0",
    "jest": "^29.4.1",
```

Then run the basic commands of `npm init @knip/config` & `npm run knip` and run into the following issue.
![image](https://github.com/user-attachments/assets/f9523a96-c4d0-4283-b923-d95e618b3077)

Here is the daemon.log

```  GNU nano 4.8                                                            ../knip/angular-monorepo/.nx/workspace-data/d/daemon.log                                                                       
[NX Daemon Server] - 2024-07-12T11:44:43.731Z - Started listening on: /tmp/1c3b5d0f404b5134dd76/d.sock
[NX Daemon Server] - 2024-07-12T11:44:43.733Z - [WATCHER]: Subscribed to changes within: ../knip/angular-monorepo (native)
[NX Daemon Server] - 2024-07-12T11:44:43.737Z - Established a connection. Number of open connections: 1
[NX Daemon Server] - 2024-07-12T11:44:43.738Z - Established a connection. Number of open connections: 2
[NX Daemon Server] - 2024-07-12T11:44:43.739Z - Closed a connection. Number of open connections: 1
[NX Daemon Server] - 2024-07-12T11:44:43.740Z - [REQUEST]: Client Request for Project Graph Received
[NX Daemon Server] - 2024-07-12T11:44:43.826Z - [REQUEST]: Updated workspace context based on watched changes, recomputing project graph...
[NX Daemon Server] - 2024-07-12T11:44:43.827Z - [REQUEST]:
[NX Daemon Server] - 2024-07-12T11:44:43.827Z - [REQUEST]:
[NX Daemon Server] - 2024-07-12T11:44:45.961Z - [REQUEST]: Responding to the client. project-graph
[NX Daemon Server] - 2024-07-12T11:44:45.963Z - Time taken for 'total for creating and serializing project graph' 2221.054794ms
[NX Daemon Server] - 2024-07-12T11:44:45.964Z - Done responding to the client project-graph
[NX Daemon Server] - 2024-07-12T11:44:45.964Z - Handled REQUEST_PROJECT_GRAPH. Handling time: 2221. Response time: 3.
[NX Daemon Server] - 2024-07-12T11:44:45.969Z - Closed a connection. Number of open connections: 0
[NX Daemon Server] - 2024-07-12T11:44:45.970Z - Established a connection. Number of open connections: 1
[NX Daemon Server] - 2024-07-12T11:44:45.984Z - Closed a connection. Number of open connections: 0

```

At which point I'm lost, not sure why the connection is being closed prematurely or something? Maybe somebody with more NX experience can shed some light on it.

**Comment by webpro** [comment]
So not sure what's tripping up the Nx daemon here. Knip does not use or interfere with it.

When you do `npm run knip` then Knip executes config files like `jest.config.js` or `vite.config.ts`. But Nx itself has only JSON config files. And without `--fix` or `--cache` Knip only reads, never writes.

**Comment by unemx** [comment]
Hello,

I've encountered the same issue with my nx repo (v18), i've upgraded to v19 to check if it would solve the issue : it didn't.
I've tried to add knip to both, a new nx project and an old project i had lying around under nx v16 : There was no issue.

I've dug a bit more and ran `yarn run knip --debug` :
```bash
[ 'nx' ]
[.] Glob options
{
  patterns: [ 'jest.config.{js,ts,mjs,cjs,json}', 'package.json' ],
  cwd: '/home/MyUser/work/MyProject',
  dir: '/home/MyUser/work/MyProject',
  gitignore: false,
  absolute: true
}
[.] Jest config file paths (1)
[ '/home/MyUser/work/MyProject/jest.config.ts' ]

 NX   Daemon process terminated and closed the connection

```

So i've tried to delete the `jest.config.ts`, and now knip is running correctly. 

----

What I tried next was to prevent Knip from scanning this file using negated pattern, but with no success.
At this point i'm unsure of why this file is scanned. Maybe it's the nx plugin ? 

In any case the workaround for now is to use `knip --production` to avoid this file.
@webpro I'm available if you want more info just lmk, thanks for this great tool :heart_hands:

**Comment by webpro** [comment]
You can try to disable the Jest plugin by setting `jest: false` in the Knip config. Or exclude that Jest config file e.g. with an empty array `"jest": { "config": [] }`

**Comment by unemx** [comment]
This work but it's still a work around. 
Why does Knip analyzing this file runs the Nx daemon ?

**Comment by webpro** [comment]
Any interesting output perhaps when using `--debug`?

**Comment by webpro** [comment]
Just had this issue myself. [Try this](https://nx.dev/concepts/nx-daemon#turning-it-off):

```
NX_DAEMON=false
```

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.35.0. See [Release 5.35.0](https://github.com/webpro-nl/knip/releases/tag/5.35.0) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._
